apps/mobile/src/components/CRUDOfferFlow/index.tsx

```diff
@@ -85,18 +96,25 @@ function CRUDOfferFlow({route: {params}, navigation}: Props): JSX.Element {
   )
 
   const onPageChange = useCallback(
-    (pageIndex: number) => {
+    (prevOrNextPageIndex: number) => {
       const currentPage =
-        screensBasedOnListingType[pageIndex] ?? 'ListingAndOfferType'
-      setCurrentStepInOfferCreation(currentPage)
-      navigation.navigate('CRUDOfferFlow', {
-        screen: currentPage,
-        offerId: params.offerId,
-      })
-      setPage(pageIndex)
+        screensBasedOnListingType[prevOrNextPageIndex] ?? 'ListingAndOfferType'
+      if (prevOrNextPageIndex < page || !dontAllowNavigationToNextStep) {
```

Comment:
IF you only need that here, you can do: 

```suggestion
      if (prevOrNextPageIndex < page || !store.get(dontAllowNavigationToNextStepAndReturnReasonAtom)) {
```